src/cmd/ksh93/tests/emacs.exp

```diff
@@ -0,0 +1,210 @@
+# Check emacs keybindings
+set pid [spawn $ksh]
+expect_prompt
+
+send "echo hello\r"
+expect "\r\nhello\r\n"
+expect_prompt 
+
+# Enable emacs keybindings
+send -- "set -o emacs\r"
+expect_prompt 
+
+# Send ctrl-p to check previous command
+send -- "\x10"
+expect -exact "set -o emacs"
+
+send -- "\x10"
+expect -exact "echo hello"
+
+# Send ctrl-n to check next command
+send -- "\xE"
```

Comment:
I'm pretty sure all chars specified using `\x` require two hexadecimal characters.

Reply:
Also, the `--` arg to stop parsing for args should be avoided unless needed, IMHO. I'm all for being explicit rather than implicit but this is one situation where the extra chars are more likely to cause the reader to ask "Why is this present?".